test/new-e2e/tests/agent-metric-logs/log-agent/windows-log/file-tailing/01_file_tailing_test.go

```diff
@@ -0,0 +1,213 @@
+func TestE2EVMFakeintakeSuite(t *testing.T) {
+	s := &WindowsFakeintakeSuite{}
+	devModeEnv, _ := os.LookupEnv("E2E_DEVMODE")
+	options := []e2e.SuiteOption{
+		e2e.WithProvisioner(awshost.Provisioner(
+			awshost.WithEC2InstanceOptions(ec2.WithOS(testos.WindowsDefault)),
+			awshost.WithAgentOptions(
+				agentparams.WithLogs(),
+				agentparams.WithIntegration("custom_logs.d", logConfig)))),
+	}
+
+	if devMode, err := strconv.ParseBool(devModeEnv); err == nil && devMode {
+		options = append(options, e2e.WithDevMode())
+	}
+	e2e.Run(t, s, options...)
+}
+
+func (s *WindowsFakeintakeSuite) BeforeTest(suiteName, testName string) {
+	s.BaseSuite.BeforeTest(suiteName, testName)
+	// Flush server and reset aggregators before the test is ran
+	utils.CleanUp(s)
+
+	// Ensure no logs are present in fakeintake before testing starts
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		logs, err := s.Env().FakeIntake.Client().FilterLogs("hello")
+		if !assert.NoError(c, err, "Unable to filter logs by the service 'hello'.") {
+			return
+		}
+		// If logs are found, print their content for debugging
+		if !assert.Empty(c, logs, "Logs were found when none were expected") {
+			cat, _ := s.Env().RemoteHost.Execute("type C:\\logs\\hello-world.log")
+			s.T().Logf("Logs detected when none were expected: %v", cat)
+		}
+	}, 2*time.Minute, 10*time.Second)
+}
+
+func (s *WindowsFakeintakeSuite) TearDownSuite() {
+	// Flush server and reset aggregators after the test is ran
+	utils.CleanUp(s)
+
+	s.BaseSuite.TearDownSuite()
+}
+
+func (s *WindowsFakeintakeSuite) TestWindowsLogTailing() {
+	// Run test cases:
+	// Given the agent configured to collect logs from a log file.
+	// When new log line is generated inside the log file.
+	// Then the agent collects the log line and forward it to the intake.
+	s.Run("LogCollection", s.LogCollection)
+	// Given the agent configured to collect logs from a log file that has no read permissions,
+	// When new log line is generated inside the log file.
+	// Then the agent fail to collects the log line.
+	s.Run("LogCollectionNoPermission", s.LogNoPermission)
+	// Given the agent configured to collect logs from a log file with reading permissions,
+	// When new log line is generated inside the log file.
+	// Then the agent collects the log line and forward it to the intake.
+	s.Run("LogCollectionAfterPermission", s.LogCollectionAfterPermission)
+	// Given the agent configured to collect logs from a log file without reading permissions and new log line actively generating,
+	// When read permission is granted
+	// Then the agent collects the log line and forward it to the intake.
+	s.Run("LogCollectionBeforePermission", s.LogCollectionBeforePermission)
+	// Given the agent configured to collect logs from a specific log file
+	// When the log file is rotated
+	// Then the agent successfully collects the log line
+	s.Run("LogRecreateRotation", s.LogRecreateRotation)
+}
+func (s *WindowsFakeintakeSuite) LogCollection() {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Can they move to not exported ? Maybe `testLogCollection`